Ahayel/email sending

artemgael96 opened the pull request.

artemgael96 committed:
add SendEmailAsync

artemgael96 committed:
Merge branch 'dev' into ahayel/email-sending

artemgael96 committed:
updated SendEmailAsync

artemgael96 committed:
updated EmailSender, Create MessageBuilder

updated EmailSender, Create MessageBuilder: add EmailSender to TestService and UserService

romanstat commented on exalearn-api/ExaLearn.Bl/Services/TestService.cs:
```diff
@@ -69,6 +73,9 @@ public async Task<AssessmentDTO> CreateAssesmentAsync(AssessmentDTO assessmentDT
             passedTest.Status = StatusType.Checked;
 
+            var user = await _userRepository.GetByIdAsync(passedTest.UserId);
+
```
var m = MessageBuilder.GenerateMessageForUserAsync(user, levelType, expirationDate, isAssing);

artemgael96 replied:
done

romanstat commented on exalearn-api/ExaLearn.Bl/Services/TestService.cs:
```diff
@@ -69,6 +73,9 @@ public async Task<AssessmentDTO> CreateAssesmentAsync(AssessmentDTO assessmentDT
 
+            var user = await _userRepository.GetByIdAsync(passedTest.UserId);
+
+            EmailSender.SendEmailAsync(user, passedTest.LevelType, passedTest.PassedTestDate, false).GetAwaiter();
```
create model in params

romanstat commented on exalearn-api/ExaLearn.Bl/EmailService/MessageBuilder.cs:
```diff
@@ -0,0 +1,62 @@
+            var message = new MailMessage(from, to)
+            {
+                Subject = isAssing ? "You are assigned to a test" : "Your a test results",
+                Body = isAssing
```
EmailBodyGenerator with 2 method return template and 1 method with switch

artemgael96 replied:
done

artemgael96 committed:
add switch case

add switch case and user  MessageBuilder.GenerateMessageForUserAsync for TestService and UserService

artemgael96 committed:
add deafault block

artemgael96 committed:
Merge branch 'dev' into ahayel/email-sending
